manim/mobject/geometry.py

```diff
@@ -529,6 +529,56 @@ def __init__(self, label, radius=None, **kwargs) -> None:
 
 
 class Ellipse(Circle):
+    """A circular shape; oval, circle.
+
+     Parameters
+     ----------
+     width : :class:`float`, optional
+        The horizontal width of the ellipse.
+     height : :class:`float`, optional
+        The vertical height of the ellipse.
+     kwargs : Any
+        Additional arguments to be passed to :class:`Circle`
+
+    Examples
+    --------
+
+    .. manim:: EllipseExample
+
+        import numpy as np
+        class EllipseExample(Scene):
+            def construct(self):
+                inc_list = []
+                ref_list = []
+
+                ellipse = Ellipse(width=2.0, height=6.5, stroke_color=WHITE, fill_color=[BLUE_A,BLUE_B], fill_opacity=1).shift(LEFT * 2)
+                focal_length = 1 / (0.49 * (1 / ellipse.width - 1 / ellipse.height))
+                end = ellipse.get_center() + RIGHT * focal_length
+
+                for angle in [x for x in np.arange(100 * DEGREES, 260 * DEGREES, 25 * DEGREES)]:
+                    lens_point = ellipse.point_at_angle(angle)
+                    incident_ray = DashedLine(
+                        start=lens_point + config.left_side,
+                        end=lens_point,
+                        stroke_width=1,
+                    )
+                    distance = end-lens_point
+                    last_value = (lens_point+(distance*2))
+
+                    refracted_ray = DashedLine(
+                        start=lens_point,
+                        end=last_value,
+                        stroke_width=1,
+                    )
+                    inc_list.append(incident_ray)
+                    ref_list.append(refracted_ray)
+
+                self.add(ellipse)
+                self.play(*[ShowCreation(inc, run_time=1.5, rate_func=linear) for inc in inc_list])
+                self.play(*[ShowCreation(ref, run_time=1.5, rate_func=linear) for ref in ref_list])
+                self.wait()
```

Comment:
Although the example is great, I don't think this needs to be a video. Considering the complexity of the example, this might be better suited for the example gallery.